packages/fxa-auth-server/scripts/audit-orphaned-customers.ts

```diff
@@ -0,0 +1,319 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import program from 'commander';
+import Stripe from 'stripe';
+import Container from 'typedi';
+import { setupProcessingTaskObjects } from '../lib/payments/processing-tasks-setup';
+import { promises as fs } from 'fs';
+import {
+  AppConfig,
+  AuthLogger
+} from '../lib/types';
+import AppError, { ERRNO } from '../lib/error';
+
+type CustomerData = {
+  id: string;
+  uid?: string;
+  email?: string;
+};
+
+type Customers = Map<string, CustomerData>
+
+const pckg = require('../package.json');
+
+const isOfStatus = (
+  keyInput: string
+): keyInput is Stripe.Subscription.Status => {
+  return [
+    'active',
+    'canceled',
+    'incomplete',
+    'incomplete_expired',
+    'past_due',
+    'trialing',
+    'unpaid',
+  ].includes(keyInput);
+};
+
+const isCustomer = (
+  customer: string | Stripe.Customer | Stripe.DeletedCustomer
+): customer is Stripe.Customer => {
+  if (typeof customer !== 'string') {
+    return !customer.deleted;
+  }
+  return false;
+};
+
+const parseDateForFirestore = (date: number) => Math.floor(date / 1000);
+
+const parseStartDate = (date: string): number =>
```

Comment:
nit. I realized while reviewing that much of this logic could be replaced by using the default value in the `commander` options. No need to change, just for future reference.